src/Draft/TranslationIndex.vue

```diff
@@ -55,4 +81,55 @@ const deleteDraft = () => {
     },
   });
 };
+
+let isSettingErrors = false;
+const isSaving = ref(false);
+
+const saveDraft = debounce(2000, () => {
+  isSaving.value = true;
+  router.post(`/draft/${props.draft.id}/save`, getPayload(), {
+    preserveScroll: true,
+    onError: () => {
+      shared.setErrors(props.errors);
+      isSettingErrors = true;
+      feedbackPanel.value.message = JSON.stringify(props.errors, null, 2);
+    },
+  });
+  isSaving.value = false;
+});
+
+const submitDraft = () => {
+  router.post(`/draft/${props.draft.id}/submit`, getPayload());
+};
+
+const publishDraft = () => {
+  if (props.user.role === 'admin') {
+    router.post(`/draft/${props.draft.id}/publish`, getPayload(), {
+      onSuccess: () => {
+        feedbackPanel.value.message = `Successfully published.`;
+        feedbackPanel.value.icon = 'check-badge';
+      },
+      onError: () => {
+        shared.setErrors(props.errors);
+        feedbackPanel.value.message = `${props.meta.chapterType} not published. Please review and correct any errors.`;
+        feedbackPanel.value.icon = 'exclamation';
+      },
+    });
+  }
+};
+
+const reject = () => {
+  // TODO: feedback
+  router.post(`/draft/${props.draft.id}/reject`, getPayload());
+};
+
+onMounted(() => {
+  model.$subscribe(() => {
+    if (isSettingErrors) {
+      isSettingErrors = false;
+      return;
+    }
```

Comment:
The develop branch has moved on from here. Please go look at the `isDirty` mechanism that disables the publish/commit buttons while the debouncer times out: 
- [Reading in the ChapterHeader](https://github.com/story-cms/ui/blob/f8ac023ec63b78cf8988793875548982187e75ab/src/Draft/DraftIndex.vue#L77)
- [Setting when saving](https://github.com/story-cms/ui/blob/f8ac023ec63b78cf8988793875548982187e75ab/src/Draft/DraftIndex.vue#L290)